### ISSUE TITLE
Sum is over both channels and groups

### ISSUE THREAD

**Issue by hjalmarlucius** [MerHS/SASA-pytorch#3]
Hi, according to the paper last sentence on pg 3 I think the sum in row 82 should only be over `fc/g` rather than all of `fc`. That should give size `b,g,fh,fw,kh,kw` in row 84.

As it currently stands, I think you effectively use `groups=1` and the spatial kernel is the same for all channels. When increasing the number of groups, I think the kernel should vary between some channels (the groups).

https://github.com/MerHS/SASA-pytorch/blob/7d113852dce2e25d4de23caf87ad7d33758c322e/model/sa_layer.py#L82

**Comment by luogen1996** [comment]
yes, i found the same problems. Meanwhile, i think the shape of  `self.relative_x` should be `(groups,self.rel_size, self.kernel_size[1])` instead of  `(self.rel_size, self.kernel_size[1])` .  In fact, when i fixed this problems, the expending  of GPU memory is almost doubled.

**Comment by chg0901** [comment]
> yes, i found the same problems. Meanwhile, i think the shape of `self.relative_x` should be `(groups,self.rel_size, self.kernel_size[1])` instead of `(self.rel_size, self.kernel_size[1])` . In fact, when i fixed this problems, the expending of GPU memory is almost doubled.

Did you mean the memory using is increased after your fix?
